### PR TITLE
Panel Meeting Date Types

### PR DESCRIPTION
**Pull request by mjoyce91**

### CODE REVIEW

File: seeds/027_panelmeetings_seed.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+const _ = require('lodash')
+const { readJson } = require('./data/helpers')
+
+const panelmeetings = readJson('./panelmeetings.json')
```
Comment (elizabeth-jimenez) [inline seeds/027_panelmeetings_seed.js:4]: needed?

File: seeds/027_panelmeetings_seed.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+              const panel_meetings = [];
+              const arr = Array(700).fill(null);
+              arr.forEach((panel, i) => {
+                let pmscode$ = 'C';
+                if (i > arr.length / 2) {
+                  pmscode$ = 'P';
+                }
+                if (i > arr.length / 4) {
+                  pmscode$ = 'I';
+                }
```
Comment (elizabeth-jimenez) [inline seeds/027_panelmeetings_seed.js:23]: Discussed in Slack: we'll never get 'P'

Comment (elizabeth-jimenez) [reply seeds/027_panelmeetings_seed.js:23]: `arr.length * .75` should work?

Comment (mjoyce91) [reply seeds/027_panelmeetings_seed.js]: I just re-ordered the logic and used `else if`

**Comment by scott062** [comment]
@mjoyce91 Do you want to merge this before we create the service to expose it?

**Comment by mjoyce91** [comment]
> @mjoyce91 Do you want to merge this before we create the service to expose it? Or do it here?

Let's merge this before creating services

**Comment by mjoyce91** [comment]
Need to update migration numbers b/c of https://github.com/MetaPhase-Consulting/mock-fsbid/pull/218

**Comment by mjoyce91** [comment]
@elizabeth-jimenez @scott062 - ready for review again

File: seeds/data/panelmeetings.json [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[
+]
```
Comment (elizabeth-jimenez) [inline seeds/data/panelmeetings.json:2]: 👀

File: migrations/057_panel_meeting_date_type.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+exports.up = function(knex) {
+  return knex.schema
+    .createTable('panelmeetingdatetypes', function(table) {
+      table.string('mdtcode').primary()
+      table.string('mdtdesctext')
+      table.string('mdtincludetimeind')
+      table.string('mdtuserinputind')
+      table.integer('mdtordernum')
```
Comment (elizabeth-jimenez) [inline migrations/057_panel_meeting_date_type.js:8]: idk if we should also add these?:
```
      "mdtcreateid": 2,
      "mdtcreatedate": "2004-03-26T00:00:00",
      "mdtupdateid": 2,
      "mdtupdatedate": "2004-03-26T00:00:00",
```

File: seeds/data/panelmeetingdates.json [hunk 5]
```diff
@@ -0,0 +1,2 @@
+[
```
Comment (elizabeth-jimenez) [inline seeds/data/panelmeetingdates.json:1]: 👀

File: seeds/data/panelmeetingdatetypes.json [hunk 6]
```diff
@@ -0,0 +1,58 @@
+[
+{
```
Comment (elizabeth-jimenez) [inline seeds/data/panelmeetingdatetypes.json:2]: https://github.com/MetaPhase-Consulting/mock-fsbid/pull/209/files#r780533087

File: seeds/028_panelmeetingdates_seed.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
+const datefns = require('date-fns');
+const { readJson } = require('./data/helpers')
+
+const panelmeetingdates = readJson('./panelmeetingdates.json')
```
Comment (elizabeth-jimenez) [inline seeds/028_panelmeetingdates_seed.js:4]: needed? If yes, i must be schooled

File: migrations/061_panel_meeting_dates.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
+exports.up = function(knex) {
+  return knex.schema
+    .createTable('panelmeetingdates', function(table) {
+      table.string('mdtcode')
+      table.integer('pmseqnum')
+
+      table.foreign('mdtcode').references('panelmeetingdatetypes.mdtcode')
+      table.foreign('pmseqnum').references('panelmeetings.pmseqnum')
+      table.datetime('pmddttm')
```
Comment (elizabeth-jimenez) [inline migrations/061_panel_meeting_dates.js:9]: do we want to add: 
`
PMD_CREATE_ID
PMD_CREATE_DATE
PMD_UPDATE_ID
PMD_UPDATE_DATE
`?

File: migrations/060_panel_meetings.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
+exports.up = function(knex) {
+  return knex.schema
+    .createTable('panelmeetings', function(table) {
+      table.integer('pmseqnum').primary()
+
+      table.string('pmscode')
+      table.string('pmtcode')
+      table.foreign('pmscode').references('panelmeetingstatuses.pmscode')
+      table.foreign('pmtcode').references('panelmeetingtypes.pmtcode')
+
```
Comment (elizabeth-jimenez) [inline migrations/060_panel_meetings.js:10]: So I know we may not want to use the other ones, but these could be helpful for when we start to tackle Panel management

```
PM_CREATE_ID
PM_CREATE_DATE
PM_UPDATE_ID
PM_UPDATE_DATE
```

File: seeds/027_panelmeetings_seed.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
+const _ = require('lodash')
+const { readJson } = require('./data/helpers')
+
+exports.seed = function(knex) {
+  return knex.raw('TRUNCATE TABLE panelmeetings CASCADE')
+    .then(function () {
+
+      return knex.select('pmscode').from('panelmeetingstatuses')
+        .then(pmscodes => {
```
Comment (elizabeth-jimenez) [inline seeds/027_panelmeetings_seed.js:9]: Needed?

File: seeds/027_panelmeetings_seed.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+          return knex.select('pmtcode').from('panelmeetingtypes')
+            .then(pmtcodes => {
+              const panel_meetings = [];
+              const arr = Array(700).fill(null);
+              arr.forEach((panel, i) => {
```
Comment (elizabeth-jimenez) [inline seeds/027_panelmeetings_seed.js:14]: just realized we are never seeding panel meeting status A - is that something we want to add?

**Comment by elizabeth-jimenez** [comment]
Will have to update migration numbers if [PR 246](https://github.com/MetaPhase-Consulting/mock-fsbid/pull/246) goes first

File: seeds/data/panelmeetings.json (hunk 3 above)
Comment (mjoyce91) [reply seeds/data/panelmeetings.json]: removed

File: seeds/028_panelmeetingdates_seed.js (hunk 7 above)
Comment (mjoyce91) [reply seeds/028_panelmeetingdates_seed.js]: not needed, removed

File: migrations/061_panel_meeting_dates.js (hunk 8 above)
Comment (mjoyce91) [reply migrations/061_panel_meeting_dates.js:9]: Will hard code

File: seeds/027_panelmeetings_seed.js (hunk 11 above)
Comment (mjoyce91) [reply seeds/027_panelmeetings_seed.js:14]: What's the A status?

Comment (elizabeth-jimenez) [reply seeds/027_panelmeetings_seed.js:14]: image
addedum

Comment (mjoyce91) [reply seeds/027_panelmeetings_seed.js:14]: updated

**Review by elizabeth-jimenez — APPROVED** [review]

**Comment by elizabeth-jimenez** [comment]
We may still want to add PM(D)_CREATE/UPDATE_ID, but idk if it needs to be in this PR

**Pull request merged**